src/solvers/flattening/boolbv_extractbits.cpp

```diff
@@ -24,50 +24,52 @@ bvt boolbvt::convert_extractbits(const extractbits_exprt &expr)
     throw 0;
   }
 
-  mp_integer o1, o2;
-  const bvt &bv0=convert_bv(expr.op0());
+  mp_integer upper_as_int, lower_as_int;
+  auto const &src_bv = convert_bv(expr.src());
 
   // We only do constants for now.
   // Should implement a shift here.
-  if(to_integer(expr.op1(), o1) ||
-     to_integer(expr.op2(), o2))
+  if(
+    to_integer(expr.op1(), upper_as_int) ||
+    to_integer(expr.op2(), lower_as_int))
     return conversion_failed(expr);
 
-  if(o1<0 || o1>=bv0.size())
+  if(upper_as_int < 0 || upper_as_int >= src_bv.size())
   {
     error().source_location=expr.find_source_location();
     error() << "extractbits: second operand out of range: "
             << expr.pretty() << eom;
   }
 
-  if(o2<0 || o2>=bv0.size())
+  if(lower_as_int < 0 || lower_as_int >= src_bv.size())
   {
     error().source_location=expr.find_source_location();
     error() << "extractbits: third operand out of range: "
             << expr.pretty() << eom;
     throw 0;
   }
 
-  if(o2>o1)
-    std::swap(o1, o2);
+  if(lower_as_int > upper_as_int)
+    std::swap(upper_as_int, lower_as_int);
```

Comment:
Not your fault, but this is somewhat weird "userfriendliness" traded for a clearly defined semantics. I'd turn this into a `DATA_INVARIANT`.

Reply:
I'd feel a bit queasy about doing that because that'd turn something that was previously valid into something that is no longer valid. It seems strange to allow this indeed, but it's not obvious to me that it's _wrong_ as such.